haystack/preview/document_stores/mixins.py

```diff
@@ -0,0 +1,30 @@
+from typing import List, Optional, Type
+
+
+from haystack.preview.document_stores.protocols import Store
+
+
+class StoreAwareMixin:
+    """
+    Adds the capability of a component to use a single document store from the `self.store` property.
+    """
+
+    _store: Optional[Store] = None
+    supported_stores: List[Type[Store]] = [Store]  # type: ignore # (see https://github.com/python/mypy/issues/4717)
+
+    @property
+    def store(self) -> Optional[Store]:
+        return self._store
+
+    @store.setter
+    def store(self, store: Store):
+        if not store:
+            raise ValueError("Can't set the value of the store to None.")
```

Comment:
We can do without this I think, the `Pipeline.add_component` already has this check.

Reply:
Well one can always set the store directly to the component, this check is preventing that. If it seems useless I can also remove it though